### PR TITLE
move suite servers to scheduler section of global config

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address #3696

They are last changes earmarked for the global config 🎉 

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests: just a renaming.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required: Documentation to be dealt with at the end of #3696

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -101,6 +101,36 @@
...
+            process. This is to prevent large numbers of suites from restarting
+            simultaneously.
+        ''')
+        Conf('ranking', VDR.V_STRING, desc='''
+            A multiline string containing Python expressions to filter
+            and/or rank hosts. For example:
+            cpu_percent() < 70
+            cpu_percent()
+            to filter by cpu_percent() < 70 then to rank by cpu_percent.
+        ''')
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:133]: Isn't this all meant to go inside `[scheduler][run hosts]`?

**Review by MetRonnie — CHANGES_REQUESTED** [review]

File: cylc/flow/cfgspec/globalcfg.py (hunk 1 above)
Comment (kinow) [reply cylc/flow/cfgspec/globalcfg.py:133]: I think it needs to have another `with Conf(...`?

**Review by kinow — APPROVED** [review]
Found nothing out of ordinary. @MetRonnie 's comment looks correct, in that the global config structure is missing a level with `run hosts`?

File: cylc/flow/cfgspec/globalcfg.py (hunk 1 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:133]: @MetRonnie  - I misread the checklist. Duh!
@kinow Not if it doesn't need to be a context for a lower level thing.

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -101,6 +101,43 @@
              desc='''
             The number of old run directory trees to retain at start-up.
         ''')
+        with Conf('run hosts', desc='''
+        Configure allowed suite hosts and ports for starting up (running or
+        restarting) suites. Additionally configure host selection settings
+        specifying how to determine the most suitable run host at any given
+        time from those configured.
+    '''):
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:109]:
```suggestion
        with Conf('run hosts', desc='''
            Configure allowed suite hosts and ports for starting up (running or
            restarting) suites. Additionally configure host selection settings
            specifying how to determine the most suitable run host at any given
            time from those configured.
        '''):
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 3]
```diff
@@ -101,6 +101,43 @@
...
+                restarted (see:ref:`auto-stop-restart`).
+            ''')
+            Conf('auto restart delay', VDR.V_INTERVAL, desc='''
+                Relates to Cylc's auto stop-restart mechanism (see
+                :ref:`auto-stop-restart`).  When a host is set to automatically
+                shutdown/restart it will first wait a random period of time
+                between zero and ``auto restart delay`` seconds before
+                beginning the process. This is to prevent large numbers of
+                suites from restarting simultaneously.
+            ''')
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:133]: Sorry, missed this in my first review - according to the proposal, this is the only one meant to be directly under `scheduler`, not `run hosts`

File: cylc/flow/cfgspec/globalcfg.py [hunk 4]
```diff
@@ -101,6 +101,43 @@
...
+    '''):
+            Conf('available', VDR.V_SPACELESS_STRING_LIST, desc='''
+                A list of allowed suite run hosts. One of these hosts will be
+                appointed for a suite to start up on if an explicit host is not
+                provided as an option to a ``run`` or ``restart`` command.
+            ''')
+            Conf(
+                'ports', VDR.V_INTEGER_LIST, list(range(43001, 43101)),
+                desc='''
+                    A list of allowed ports for Cylc to use to run suites.
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:118]: For consistency
```suggestion
            Conf('ports', VDR.V_INTEGER_LIST, list(range(43001, 43101)),
                 desc='''
                A list of allowed ports for Cylc to use to run suites.
```

**Review by MetRonnie — CHANGES_REQUESTED** [review]

**Review by MetRonnie — APPROVED** [review]

**Comment by MetRonnie** [comment]
I'll leave it you Tim whether you want to merge or squash/rebase

**Pull request merged**